packages/langchain/src/openai/chat.ts

```diff
@@ -0,0 +1,74 @@
+import { CallbackManagerForLLMRun } from '@langchain/core/callbacks/manager';
+import { BaseMessage } from '@langchain/core/messages';
+import type { ChatResult } from '@langchain/core/outputs';
+import { AzureChatOpenAI, AzureOpenAI } from '@langchain/openai';
+import { OpenAiChatClient as OpenAiChatClientBase } from '@sap-ai-sdk/foundation-models';
+import { mapLangchainToAiClient, mapResponseToChatResult } from './util.js';
+import type { OpenAiChatModelInput, OpenAiChatCallOptions } from './types.js';
+
+/**
+ * OpenAI Language Model Wrapper to generate texts.
+ */
+export class OpenAiChatClient extends AzureChatOpenAI {
+  declare CallOptions: OpenAiChatCallOptions;
```

Comment:
[q] why not  `callOptions`, also is it meant to be public?

Reply:
CallOptions is a property from the superclass, by declaring it I only give it its proper type.
I guess I could overwrite the visibility from the superclass, but I'm not sure if that's a good idea, what do you think?

🤔

Reply:
No, go with the standard. But is this the standard way to give it a type? If not, why do we need a different type?